apps/src/util/curriculumRecommender/curriculumRecommender.js

```diff
@@ -109,6 +107,87 @@ export const getSimilarRecommendations = (
   return sortRecommendations(curriculaScores).map(curr => curr[0]);
 };
 
+// Stretch Curriculum Recommender:
+//    1) Filters out the curriculum with the a key of mainCurriculumKey, any curricula that do not support any of mainCurriculum's grade levels,
+//        and (if applicable) any curricula the user has taught before
+//    2) Scores the remaining curricula to find which is the best stretch fit for the mainCurriculum
+export const getStretchRecommendations = (
+  curriculaData,
+  mainCurriculumKey,
+  curriculaTaught
+) => {
+  // Create copy of array of curricula to not affect the curriculaData array passed in
+  let curricula = [...curriculaData];
+
+  // Filter out the curriculum with the key, mainCurriculumKey, from the curricula to consider
+  const mainCurriculumIndex = curricula.findIndex(
+    e => e.key === mainCurriculumKey
+  );
+  const mainCurriculum = curricula.splice(mainCurriculumIndex, 1)[0];
+
+  // Filter out curricula that don't support any of the same grade levels
+  curricula = filterHasOverlappingGradeLevels(
+    curricula,
+    mainCurriculum.grade_levels
+  );
+
+  // (if signed-in teacher) Filter out curricula the user has taught before
+  if (curriculaTaught) {
+    curricula = filterNotTaughtBefore(curricula, curriculaTaught);
+  }
+
+  // Score each remaining curricula
+  const curriculaScores = [];
+  curricula.forEach(curriculum => {
+    let score = 0;
+
+    // Add points if given curriculum has a slighty longer duration.
+    score += hasSlightlyLongerDuration(curriculum, mainCurriculum.duration)
+      ? STRETCH_RECOMMENDER_SCORING['hasDesiredDuration']
+      : 0;
+
+    // Add points if given curriculum has a DIFFERENT marketing initiative.
+    score += hasDesiredMarketingInitiative(
+      curriculum,
+      mainCurriculum.marketing_initiative
+    )
+      ? 0
+      : STRETCH_RECOMMENDER_SCORING['hasDesiredMarketingInitiative'];
+
+    // Adds points for having NO school subject. If it has school subjects, then add
+    // points if it's NOT overlapping with the mainCurriculum's subjects.
+    if (!hasAnySchoolSubject(curriculum)) {
+      score += STRETCH_RECOMMENDER_SCORING['hasAnySchoolSubject'];
+    } else {
+      const numOverlappingSubjects = numOverlappingDesiredSchoolSubjects(
+        curriculum,
+        mainCurriculum.school_subject
+      );
+      score +=
+        numOverlappingSubjects === 0
+          ? STRETCH_RECOMMENDER_SCORING['overlappingDesiredSchoolSubject']
+          : 0;
+    }
+
+    // Add points if given curriculum has an important topic.
```

Comment:
are subject and topic different?

Reply:
Yep subject is connected to school subjects like math and science while topic would be things like AI or App Design